strapi/Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM node:18-alpine
+# Installing libvips-dev for sharp Compatibility
+RUN apk update && apk add --no-cache build-base gcc autoconf automake zlib-dev libpng-dev nasm bash vips-dev
+ARG NODE_ENV=development
+ENV NODE_ENV=${NODE_ENV}
```

Comment:
What do you think about using [multi-stage builds](https://docs.docker.com/build/building/multi-stage/) ? IMO this would be better, because the build tools would stay in the builder step image. Then we only transfer the required compiled assets and have a final image that is minimal and that only contains the runtime and the executed code.

Reply:
I'm not sure if this would be worth the trouble of separating the build and runtime dependencies; as I said below, this is the provided `Dockerfile` for strapi

Reply:
I see, in any case we can use this for now and improve later. IMO using multi-stage builds have major benefits, such as slimmer images (hence faster downloads on upate) as well as a smaller attack surface (the container does not provide these executables in case of e.g. an injection breach). But I agree we should consider if it's a pain to define the required set of files.

Reply:
I've tried a bit to use a multi-stage build, but I ran into some errors because of missing dependencies. To my understanding, the build creates dependencies outside of the `WORKDIR`. Furthermore, for the source and build directories, their size is insignificant compared to `node_modules` (which is obviously required), so this will not reduce the final image size.

I don't think we can get significant benefits from a multi-stage build

Reply:
Thank you for your research. This seems odd, since theoretically only the built artifacts compiled from the source ts files as well as the js dependencies in `node_modules` should be required, alongside the `node` runtime. Let's forget this for now and I'll open an issue to investigate this further.